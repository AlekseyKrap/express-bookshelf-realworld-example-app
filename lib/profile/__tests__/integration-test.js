const request = require('supertest');
const app = require('../../app');

describe('profile integration', () => {
  describe('GET /api/profiles/:username', () => {
    describe('with invalid username', () => {
      it('should return a 404', async () => {
        const response = await request(app)
          .get('/api/profiles/invalid')
          .set('Accept', 'application/json')
          .send();

        expect(response.statusCode).toBe(404);
        expect(response.body).toMatchSnapshot();
      });
    });
  });
});
