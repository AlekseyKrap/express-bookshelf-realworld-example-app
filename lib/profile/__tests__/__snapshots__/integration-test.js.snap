// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile integration GET /api/profiles/:username with invalid username should return a 404 1`] = `
Object {
  "errors": Object {
    "error": Object {
      "data": null,
      "isBoom": true,
      "isServer": false,
      "message": "Not Found",
      "output": Object {
        "headers": Object {},
        "payload": Object {
          "error": "Not Found",
          "message": "Not Found",
          "statusCode": 404,
        },
        "statusCode": 404,
      },
    },
    "message": "Not Found",
  },
}
`;
